docs/dev/hacking-guide.md

```diff
@@ -0,0 +1,177 @@
+# Cluster cloud controller manage operator - Hacking guide
+
+## How to run a component locally for testing
+
+Prerequisites:
+
+```bash
+git checkout github.com/openshit/$repository_name
+cd $repository_name
+```
+
+Make sure your $KUBECONFIG is set properly, because it will be used to interact with your cluster.
+
+First step is to scale down the cluster version operator.
+
+“At the heart of OpenShift is an operator called the Cluster Version Operator. This operator watches the deployments and images related to the core OpenShift services, and will prevent a user from changing these details. If I want to replace the core OpenShift services I will need to scale this operator down.”
+
+```bash
+oc scale --replicas=0 deployment/cluster-version-operator -n openshift-cluster-version
+```
+
+Second step is scaling down the CCCMO. The operator watches for deployed resources to be running and will prevent them from scaling down.
+
+```bash
+oc scale deployment/cluster-cloud-controller-manager -n openshift-cloud-controller-manager-operator --replicas=0
+```
+
+Finally, once all has been scaled down you can compile and run the controller.
+
+You can copy or edit `hack/example-images.json` with the provider image you want to run.
+
+```bash
+make build
+ ./bin/cluster-cloud-controller-manager --images-json=hack/example-images.json
+```
+
+## How to build the operator in a container for remote testing
+
+Prerequisites:
+
+```bash
+git checkout github.com/openshit/$repository_name
+cd $repository_name
+```
+
+First you’ll need to build an image with your changes.
+
+```bash
+IMG="quay.io/your-repo/cluster-cloud-controller-manager-operator:latest" make images
+docker push quay.io/your-repo/cluster-cloud-controller-manager-operator:latest
```

Comment:
I'd suggest instead of using the latest tag, we suggest to users to tag their work with something meaningful, eg their branch name?
```suggestion
IMG="quay.io/your-repo/cluster-cloud-controller-manager-operator:<branch-name>" make images
docker push quay.io/your-repo/cluster-cloud-controller-manager-operator:<branch-name>
```